impl/ocaml/sqlgg_runtime.ml

```diff
@@ -0,0 +1,10 @@
+
+let replace_all_chars ~str ~sub ~by =
+  let rec get_chunks s sub acc off =
+    if off >= String.length s then List.rev ("" :: acc)
+    else
+      match String.index_from_opt s off sub with
+      | None -> List.rev (String.sub s off (String.length s - off) :: acc)
+      | Some off' -> get_chunks s sub (String.sub s off (off' - off) :: acc) (off' + 1)
+  in
```

Comment:
this reimplements String.split_on_char which is in stdlib, why

Reply:
The real reason is that I forgot it. The post hoc one is that `split_on_char` is 4.03+. Is 4.02 still supported?

Reply:
I am happy to drop 4.02 in favor of split_on_char :muscle: